lib/utils/resolveSilent.mjs

```diff
@@ -15,6 +15,15 @@ const pathSuffixes = ['', '.js', '.json', `${path.sep}index.js`, `${path.sep}ind
 
 const specifierSuffixes = ['', '.js', '.json', '/index.js', '/index.json'];
 
+/**
+ * Checks whether the given file exists as a file.
+ * @param {string} filename The file to check.
+ * @returns {boolean} `true` if the file exists and is a file, otherwise `false`.
+ */
+function existsFile(filename) {
+	return fs.existsSync(filename) && fs.statSync(filename).isFile();
```

Comment:
[suggest] Minor. How about using the `throwIfNoEntry` option like this? I can reduce one `fs` call.

```suggestion
	return fs.statSync(filename, { throwIfNoEntry: false })?.isFile() ?? false;
```